client/src/reducers/auth.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  REGISTER_SUCCESS,
+  REGISTER_FAILURE,
+  LOGIN_SUCCESS,
+  LOGIN_FAILURE,
+  LOG_OUT,
+  AUTH_ERROR,
+  USER_LOADED,
+} from "../actions/types";
+
+interface InitialState {
+  token: string | any;
+  isAuthenticated: boolean;
+  loading: boolean;
+  user: null | any;
+}
+
+const initialState: InitialState = {
+  token: localStorage.getItem("token"),
+  isAuthenticated: false,
+  loading: true,
+  user: null,
+};
+
+export default function (state = initialState, action: any): any {
+  const { payload, type } = action;
+  switch (type) {
+    case REGISTER_SUCCESS:
+    case LOGIN_SUCCESS:
+      localStorage.setItem("token", payload.token);
+      return {
```

Comment:
Remove any side effects from reducers, they should be a clear functions

Reply:
done